Solution1.java

```diff
@@ -0,0 +1,34 @@
+class Solution1 {
+    // Time :  O(logn)  Space : O(1)
+    public int search(int[] nums, int target) {
+        
```

Comment:
Space is not constant as you are recursing

Reply:
Thanks, Kannan I might need to read more about the complexity